src/ERC4973Permit.sol

```diff
@@ -64,7 +64,12 @@ abstract contract ERC4973Permit is ERC4973, EIP712, IERC4973Permit {
     string calldata tokenURI
   ) internal view returns (bytes32) {
     bytes32 structHash = keccak256(
-      abi.encode(MINT_PERMIT_TYPEHASH, from, to, tokenURI)
+      abi.encode(
```

Comment:
what if we used `abi.encodePacked` here? Is it equal to your proposed change?

Reply:
Tried it with the following and the hashes will not match, but that's expected because EIP specifies that these need to be exactly 32 bytes. 

```solidity
      abi.encodePacked(
        MINT_PERMIT_TYPEHASH,
        from,
        to,
        tokenURI
      )
```

Here's relevant info from EIP 
https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct

```
Definition of encodeData
The encoding of a struct instance is enc(value₁) ‖ enc(value₂) ‖ … ‖ enc(valueₙ), i.e. the concatenation of the encoded member values in the order that they appear in the type. Each encoded member value is exactly 32-byte long.
...
...
The dynamic values bytes and string are encoded as a keccak256 hash of their contents.
...
...
```

From solidity docs on `abi.encodePacked`. 
https://docs.soliditylang.org/en/v0.8.15/abi-spec.html?highlight=abi.encode#strict-encoding-mode